cpp/tests/replace/replace_nulls_tests.cpp

```diff
@@ -158,7 +158,7 @@ TEST_F(ReplaceNullsStringsTest, ReplaceNone)
   std::unique_ptr<cudf::column> result;
   ASSERT_NO_THROW(result = cudf::replace_nulls(input_w, replacement_w, mr()));
 
-  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*result, expected_w);
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*result, expected_w);
```

Comment:
Why this change?

Reply:
The test was failing because it expected there to be no null mask on the result column, but after the optimization it can be there.  That's why I changed it to check for equivalence testing.

When copying the string with a full gather, the null count is computed and the null mask can optimized out, but when copying the string view via just buffer copies, it does not compute the null count and instead copies the bitmask range of the input if the input has a mask.  In this test case the input does have a mask even though the null count is zero, so the output also has a mask.

If we want to optimize out the null mask, the code can compute the null count on the input string view and only copy the incoming view's null mask if there are any nulls.  That will change the code to synchronize with the GPU, but it could save a validity mask alloc and copy in the case where there are no nulls.  If that's desired I can update, and this test change will be removed.

Reply:
I don't have an opinion on this.

Reply:
Yeah I similarly don't have a strong opinion here.  In some cases it will be better to avoid the null count, and sometimes it will be worth the memory savings.

Is this ready to go as-is, or do we need to find more opinions on this?